pkg/arvo/sys/zuse.hoon

```diff
@@ -2373,6 +2376,19 @@
             [%disavow ~]
         ==  ==
       ::
+      ++  udiff-to-diff
+        |=  [=a=udiff =a=point]
+        ^-  diff
+        ?-    +<.a-udiff
+            %disavow  ~|(%udiff-to-diff-disavow !!)
```

Comment:
I take it this is the handler for a sponsor disavowing its sponsee... Is this a todo for later, or is there a reason it's ok to stub out this case here?

Reply:
Well, I guess it's a conversion function, not a handler, but I guess I don't fully understand the semantics of the invertible vs. non-invertible diffs or where they come from.

Reply:
A `diff` is invertible, a `udiff` is not.  What we get from the chain is not directly invertible without knowing some state about the point.  For example, seeing "new keys at life 5 with values X,Y" is not invertible unless we already know what the keys were at previous lives.

Azimuth-tracker doesn't maintain state for each point; Kale does.  Azimuth-tracker sends `udiff`s to Kale, which converts them to `diff`s for internal use and to send to other Kales.

Disavowing a block says "I detected that this block that I already told you about was reorganized away".  But the azimuth-tracker doesn't know enough to give any more information than that, so Kale needs to handle that.  But this function can't handle that because you need more state than `point` to do that correctly.

Reply:
Huh, ok.  Where does Kale handle converting reorg udiff into a diff, or is that not implemented yet?

Reply:
Not implemented yet, but it would be in `+new-event`.  I may not implement it for a while, because it should only get triggered if we have a reorg greater than 30 blocks, which is really rare.  But it's important we recognize when it happens.